sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderClient.cs

```diff
@@ -0,0 +1,522 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.ServiceBus.Core;
+using Azure.Messaging.ServiceBus.Diagnostics;
+using Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.ServiceBus.Sender
+{
+    /// <summary>
+    ///   A client responsible for publishing <see cref="ServiceBusMessage" /> to a specific Service Bus entity,
+    ///   grouped together in batches.  Depending on the options specified when sending, events data
+    ///   may be automatically routed to an available partition or sent to a specifically requested partition.
+    /// </summary>
+    ///
+    /// <remarks>
+    ///   Allowing automatic routing of partitions is recommended when:
+    ///   <para>- The sending of events needs to be highly available.</para>
+    ///   <para>- The event data should be evenly distributed among all available partitions.</para>
+    ///
+    ///   If no partition is specified, the following rules are used for automatically selecting one:
+    ///   <para>1) Distribute the events equally amongst all available partitions using a round-robin approach.</para>
+    ///   <para>2) If a partition becomes unavailable, the Service Bus service will automatically detect it and forward the message to another available partition.</para>
+    /// </remarks>
+    ///
+    public class ServiceBusSenderClient : IAsyncDisposable
+    {
+        /// <summary>The minimum allowable size, in bytes, for a batch to be sent.</summary>
+        internal const int MinimumBatchSizeLimit = 24;
```

Comment:
Where are we using `MinimumBatchSizeLimit` ?

Reply:
Removed